test/BDNPerfTests/run_bdnperftest.ps1

```diff
@@ -0,0 +1,577 @@
+            Write-Host "** "
+        }
+        return $false # the values are too different
+    }
+  }
+
+######### ParseValueFromResults ###########
+#  
+# Takes the line from the results file and returns the value from the requested column
+# strips off all the characters and just to return the actual value
+#
+# NOTE: Example of ResultsLine from BDN benchmark: "| InlinePing   |   2.343 us | 0.0135 us | 0.0113 us |         - |"
+# NOTE: columnNum is zero based
+#
+######################################################
+function ParseValueFromResults {
+param ($ResultsLine, $columnNum)
+
+    # Remove the leading and trailing pipes and split the string by '|'
+    $columns = $ResultsLine.Trim('|').Split('|') 
+    $column = $columns | ForEach-Object { $_.Trim() }
+    $foundValue = $column[$columnNum] 
+    if ($foundValue -eq "-") {
+        $foundValue = "0"
+    }
+   
+    $foundValue = $foundValue.Trim(' B')  
+    $foundValue = $foundValue.Trim(' K')  
+    $foundValue = $foundValue.Trim(' m')  
+
+    return $foundValue
+}
+
+
+# ******** BEGIN MAIN  *********
+# Get base path since paths can differ from machine to machine
+$pathstring = $pwd.Path
+if ($pathstring.Contains("test")) {
+    $position = $pathString.IndexOf("test")
+    $basePath = $pathstring.Substring(0,$position-1)  # take off slash off end as well
+} else {
+    $basePath = $pathstring  # already in base as not in test
+    Set-Location .\test\BDNPerfTests\
+}
+
+# Add .json to end if not already there
+if ($configFile -notlike "*.json") {
+    $configFile += ".json"
+}
+
+# This is special case that allows passing test without specifying CI_CONFIG_BDN_ at the beginning - need for perf test
+if ($configFile -notlike "CI_CONFIG_BDN_Benchmark_*") {
+    $configFile = "CI_CONFIG_BDN_Benchmark_" + $configFile  
+}
+
+# Read the test config file and convert the JSON to a PowerShell object
+$fullConfiFileAndPath = "ConfigFiles/$configFile"
+if (-not (Test-Path -Path $fullConfiFileAndPath)) {
+    Write-Error -Message "The test config file $fullConfiFileAndPath does not exist." -Category ObjectNotFound
+    exit
+}
+$json = Get-Content -Raw $fullConfiFileAndPath
+$object = $json | ConvertFrom-Json
+
+# Use this in the file name to separate outputs when running in ADO
+$CurrentOS = "Windows"
+if ($IsLinux) {
+    $CurrentOS = "Linux"
+}
+
+Write-Host "************** Start BDN.benchmark Test Run ********************" 
+Write-Host " "
+
+# Set all the config options (args to benchmark app) based on values from config json file
+$configuration = $object.configuration
+$framework = $object.framework
+$filter = $object.filter
+$allocatedColumn = "-1"   # last one is allocated, just to ensure in case other column gets added
+$paramNone = "None"
+$paramDSV = "DSV"
+$paramACL = "ACL"
+$paramAOF = "AOF"
+
+# Set the expected values based on the OS
+if ($IsLinux) {
+    # Linux expected values
+    $expectedGet_None_AllocatedValue = $object.expectedGET_None_AllocatedValue_linux
```

Comment:
this isn't maintainable. why not iterate over the retreived JSON object fields to find what we need instead of expressing each one here.

Reply:
I could do a switch statement based on what test is running and only get those values, but I was thinking it would just be easier to run through them all. Having the switch statement would clearly define what is expected from each specific test.

Reply:
Turns out the switch on config file to only read values we need for that specific tests looks better and is more concise.